src/components/DateRangePicker.jsx

```diff
@@ -104,20 +105,26 @@ export default class DateRangePicker extends React.Component {
       this.props.onFocusChange(END_DATE);
 
       startDate = day;
+
       if (isInclusivelyAfterDay(day, endDate)) {
         endDate = null;
       }
     } else if (focusedInput === END_DATE) {
-      if (isInclusivelyBeforeDay(day, startDate)) {
-        startDate = day;
-        endDate = null;
-      } else {
+      const firstAllowedEndDate = startDate && startDate.clone().add(minimumNights, 'days');
+
+      if (!startDate) {
+        endDate = day;
+        this.props.onFocusChange(START_DATE);
+      } else if (isInclusivelyAfterDay(day, firstAllowedEndDate)) {
         endDate = day;
         if (!startDate) {
           this.props.onFocusChange(START_DATE);
```

Comment:
This case will never happen because it'll be cause by line 115. We should delete this.